File: OLSource1/v1/VS_officedev/codesnippet/CSharp/converting-between-sharepoint-project-system-types-and-other-visual-studio-project-types_1.cs
        void projectService_ProjectAdded(object sender, Microsoft.VisualStudio.SharePoint.SharePointProjectEventArgs e)
        {
            EnvDTE.Project dteProject = e.Project.ProjectService.Convert<
                Microsoft.VisualStudio.SharePoint.ISharePointProject, EnvDTE.Project>(e.Project);

            if (dteProject != null)
            {
                // Use the Visual Studio automation object model to add a folder to the project.
                dteProject.ProjectItems.AddFolder("Data");
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/converting-between-sharepoint-project-system-types-and-other-visual-studio-project-types_1.vb
    Private Sub projectService_ProjectAdded(ByVal sender As Object, _
        ByVal e As Microsoft.VisualStudio.SharePoint.SharePointProjectEventArgs)

        Dim dteProject As EnvDTE.Project = e.Project.ProjectService.Convert( _
            Of Microsoft.VisualStudio.SharePoint.ISharePointProject, EnvDTE.Project)(e.Project)
        If dteProject IsNot Nothing Then
            ' Use the Visual Studio automation object model to add a folder to the project.
            dteProject.ProjectItems.AddFolder("Data")
        End If
    End Sub